extensions-core/stats/src/main/java/org/apache/druid/query/aggregation/variance/VarianceAggregatorFactory.java

```diff
@@ -239,7 +240,9 @@ public Comparator getComparator()
   @Override
   public Object finalizeComputation(@Nullable Object object)
   {
-    return object == null ? null : ((VarianceAggregatorCollector) object).getVariance(isVariancePop);
+    return object == null || ((VarianceAggregatorCollector) object).count == 0
+           ? null
```

Comment:
I think so. I was able to hit this bug because of `VarianceSqlAggregatorTest#testVarianceOrderBy` One of the columns had no value for a row that was being grouped on.

I then read this comment in `getVariance(...)` 

```
// in SQL standard, we should return null for zero elements. But druid there should not be such a case
```
which is why I returned null.

I guess it could be nicer to return `NullHandling.defaultDoubleValue()` instead

Reply:
Ah, that comment is out of date. It used to be true but now timeseries queries can return a single row, SQL-style, even if there's no input data. There's a test for it in TImeseriesQueryRunnerTest -> testEmptyTimeseries. It looks like it's expecting `NullHandling.defaultDoubleValue()` for the aggregators it tests.

Reply:
I approved the patch, but please also change the comment to become accurate. Perhaps also include a test like the empty timeseries one I mentioned.

Reply:
👍 I'll add the test and push up a new patch